components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -315,6 +332,7 @@ IJsonSerializer jsonSerializer
                 // Update local state.
                 _availableJobs = availableJobs;
                 _assignedJobs = assignedJobs;
+                _publisherDiagnosticInfo = publisherDiagnosticInfo;
```

Comment:
We should also carry over diagnostics info for the jobs that remained. Meaning that whenever we find an assigned job that should remain (line `324: assignedJobs[jobIdDict[newJobId].Item2] = newJob`) we should also add corresponding diagnostics info to `publisherDiagnosticInfo`.

Reply:
this is already done in `SendHeartbeatAsync` in

`foreach (var assignedJob in _assignedJobs) {
                        if (info.Id == assignedJob.Value.Job.Id) {
                            _publisherDiagnosticInfo.AddOrUpdate(assignedJob.Value.Job.Id, info);
                        }
                    }`
for every jobs in assignedJob, the diagnostic info is added or updated